sw/device/lib/base/mmio.h

```diff
@@ -234,6 +234,30 @@ inline void mmio_region_nonatomic_set_mask32(mmio_region_t base,
   mmio_region_write32(base, offset, value);
 }
 
+/**
+ * Sets the field |value| from the MMIO region |base| at the given offset.
+ *
+ * This function performs a non-atomic read-write-modify operation on a
+ * MMIO region.
+ *
+ * @param base the region to set the field in.
+ * @param offset the offset to set the field at, in bytes.
+ * @param mask the field mask.
+ * @param index field position within the selected register.
+ * @param value the field value to be written at the @p index position within
+ *              the selected register.
+ */
+inline void mmio_region_nonatomic_set_field32(mmio_region_t base,
+                                              ptrdiff_t offset,
+                                              uint32_t mask,
+                                              uint32_t index,
+                                              uint32_t value) {
```

Comment:
Index and value could be passed inside a structure, that probably would align the API better with potential @mcy suggested changes to the reggen, see #1710 .

Reply:
Done.